controllers/entities.js

```diff
@@ -30,9 +30,10 @@ exports.create = function(req, res, next) {
     if (result.errors > 0) return next(result.first_error);
     return result.changes[0].new_val;
   })
-  .then(entity =>
-    fanout.resolvePermissions(entity.id, req.body.permissions)
-    .then(() => res.send(entity)))
+  .then(entity => fanout.resolvePermissions(entity.id, req.body.permissions)
+    .then(() => r.table('entities').get(entity.id))
+    .then(entity => res.send(entity))
+  )
   .then(next)
```

Comment:
isn't it going to implicitly return `res.send(entity)`
this will then get passed into .then(next)
and then next is a callback with an error as it's first argument won't next be called as an error
... i also may be completely wrong...